src/NUnitTestAdapter/NUnit3TestExecutor.cs

```diff
@@ -266,7 +266,13 @@ private void RunAssembly(string assemblyName, TestFilter filter)
                     }
                 }
                 else
-                    TestLog.Info("NUnit failed to load " + assemblyName);
+                {
+                    var msgNode = loadResult.SelectSingleNode("properties/property[@name='_SKIPREASON']");
+                    if (msgNode != null && (new[] { "contains no tests", "Has no TestFixtures" }).Any(msgNode.GetAttribute("value").Contains))
+                        TestLog.Info("Assembly " + assemblyName + " contains no tests supported by this runner: " + _activeRunner.GetType().AssemblyQualifiedName);
```

Comment:
Maybe, but it is not quite correct. In my case it was a mismatch between 3.4 and 3.5 that caused the fault (thats why I returned the full AssemblyQualifiedName in the message)